src/components/shapes/draw_newshape/defaults.js

```diff
@@ -1,8 +1,25 @@
 'use strict';
 
 var Color = require('../../color');
+var Lib = require('../../../lib');
 
 
+function dfltLabelYanchor(isLine, labelTextPosition) {
+    var dfltYanchor;
+    if(isLine) {
+        dfltYanchor = 'bottom';
```

Comment:
You may simply `return` these values without declaring `dfltYanchor`.
Also you may avoid if statement blocks by using `?` operator.

```js
function dfltLabelYanchor(isLine, labelTextPosition) {
    return isLine ? 'bottom' :
        labelTextPosition.indexOf('top') !== -1 ? 'top' :
        labelTextPosition.indexOf('bottom') !== -1 ? 'bottom' : 'middle';
}
```

Reply:
Roger, wasn't sure how to capture this logic using the `?` operator but what you wrote makes sense.